### PR TITLE
Enable use of <optional_time_frame> argument

### PR DESCRIPTION
**Pull request by DanielMwendwa**

### CODE REVIEW

**Comment by DanielMwendwa** [comment]
> 1. What is the difference between time frame and optional time frame?
> 2. Why would you want to set one but not the other, or a different value for each?

1. The difference is that one is used instead of a constant timeframe. This is actually for testing purposes. i.e increasing the timeframe to reduce the script execution time.
2. When computing message difference between two firebase time periods - the time period for firebase is a constant number which is 10 minutes. The optional time frame is set to overwrite this constant. The time frame value is used to set an interval between the given start and end dates for computing the number of messages.

File: mno_analysis_tools/run-mno-analysis.sh [hunk 1]
```diff
@@ -38,5 +52,10 @@ echo "Starting run with id '$RUN_ID'"
 ./docker-run-compute-messages-per-period.sh --profile-memory ./data "$RAW_MESSAGES_FILE_PATH" "$TARGET_OPERATOR" \
     "$TARGET_MESSAGE_DIRECTION" "$START_DATE" "$END_DATE" "$TIME_FRAME" "$OUTPUT_DIR"
 
-./docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh --profile-memory ./data "$RAW_MESSAGES_FILE_PATH" \
+if [[ "$OPTIONAL_TIME_FRAME" = true ]]; then
+    ./docker-run-compute-msg-difference-btwn-two-firebase-time-periods.sh --profile-memory ./data "$RAW_MESSAGES_FILE_PATH" \
+    "$TARGET_OPERATOR" "$TARGET_MESSAGE_DIRECTION" "$START_DATE" "$END_DATE" "$OUTPUT_DIR" "$OPTIONAL_TIME_FRAME_VALUE"
```
Comment (as2388) [inline mno_analysis_tools/run-mno-analysis.sh:57]: If this is optional move it to an optional flag instead. See https://github.com/AfricasVoices/Project-WorldVision/blob/master/run_scripts/5_automated_analysis.sh for an example of how to accept an optional arg to this script and forward it into another shell script more cleanly, without needing an if.

**Comment by as2388** [comment]
> 1. The difference is that one is used instead of a constant timeframe. This is actually for testing purposes. i.e increasing the timeframe to reduce the script execution time.
> 2. When computing message difference between two firebase time periods - the time period for firebase is a constant number which is 10 minutes. The optional time frame is set to overwrite this constant. The other time frame value is used to set an interval between the given start and end dates for computing the number of messages.

I'm still not convinced. Whether you're looking at messages per time period or the difference between time periods, I don't see why they would need to be different for the same run. If they do, please change the names to describe which time periods they are referring to, rather than whether they happen to be optional or not. Happy to discuss more offline.

File: mno_analysis_tools/run-mno-analysis.sh (hunk 1 above)
Comment (DanielMwendwa) [reply mno_analysis_tools/run-mno-analysis.sh]: Thanks

**Pull request merged**